payments/utils.py

```diff
@@ -365,7 +365,7 @@ def send_payment_notification(order, request, email=None):
 
     order_email = email or order.customer_email
 
-    if not order_email:
+    if not order_email or "@example.com" in order_email:
```

Comment:
The same check is done in invoice.py, maybe refactor into common utility and use in both places

def is_missing_email(email_address):
    return not email_address or "@example.com" in email_address